src/Classes/TradeQueryCurItem.lua

```diff
@@ -0,0 +1,243 @@
+	easy:setopt_useragent("Path of Building/" .. launch.versionNumber .. "test")
+	easy:setopt_writefunction(function(data)
+		tradeStats = tradeStats..data
+		return true
+	end)
+	easy:perform()
+	easy:close()
+	return tradeStats
+end
+
+-- hit the api to get a list of items, need this to determine if items are accessories or not
+function TradeQueryCurItem:FetchItems()
+	local items = ""
+	local easy = common.curl.easy()
+	easy:setopt_url("https://www.pathofexile.com/api/trade/data/items")
+	easy:setopt_useragent("Path of Building/" .. launch.versionNumber .. "test")
+	easy:setopt_writefunction(function(data)
+		items = items..data
+		return true
+	end)
+	easy:perform()
+	easy:close()
+	return dkjson.decode(items)
+
+end
+
+-- template for getting the url endpoint
+function TradeQueryCurItem:GetTemplate(type, attribuiteIDs)
+    local template = {
+        ["query"] = {
+          ["status"] = {
+            ["option"] = "online"
+          },
+          ["type"] = type,
+          ["stats"] = {
+            {
+              ["type"] = "and",
+              ["filters"] = attribuiteIDs
+            }
+        },
+          ["filters"] = {
+            ["type_filters"] = {
+      
+            }
+          }
+        },
+        ["sort"] = {
+          ["price"] = "asc"
+        }
+      }
+    return dkjson.encode(template)
+end
+
+-- passes the template to recieve the endpoint
+function TradeQueryCurItem:GetTradeEndpoint(template)
+    local tradeLink = ""
+	local easy = common.curl.easy()
+	easy:setopt_url("https://www.pathofexile.com/api/trade/search/" .. self.leauge)
+	easy:setopt_useragent("Path of Building/" .. launch.versionNumber .. "test")
+    local header = {"Content-Type: application/json"}
+    easy:setopt(common.curl.OPT_HTTPHEADER, header)
+    easy:setopt(common.curl.OPT_POST, true)
+    easy:setopt(common.curl.OPT_POSTFIELDS, template)
+    easy:setopt(common.curl.OPT_ACCEPT_ENCODING, "")
+	easy:setopt_writefunction(function(data)
+		tradeLink = tradeLink..data
+		return true
+	end)
+	easy:perform()
+	easy:close()
+    return tradeLink
+end
+
+-- appends the endpoint to the default trade url
+function TradeQueryCurItem:GetTradeUrl(endpoint)
+    return 'https://www.pathofexile.com/trade/search/'.. self.leauge .. '/' .. endpoint;
+end
+
+function TradeQueryCurItem:GetFileCreatedTime(path)
+    local cwd = io.popen("cd"):read()
+    path = path:gsub('/', '\\')
+    local cdate = io.popen( "dir /T:C \""..cwd..path.."\"", "r" )
+    local createdate = cdate:read("*all")
+    local pos = 0
+    
+    for i = 0, 4 do
+       pos = string.find(createdate, "\n", pos+1)
+    end
+    
+    createdate = string.sub(createdate, pos, pos+17)
+    local year = string.sub(createdate, 8, 11)
+    local month = string.sub(createdate, 5, 6)
+    local day = string.sub(createdate, 2, 3)
+    local hour = string.sub(createdate, 14, 15)
+    local minute = string.sub(createdate, 17, 18)
+    
+    return month.."-"..day
+end
```

Comment:
This code seems to only be used in a commented out line. Please remove if not necessary

```Lua
local cdate = io.popen( "dir /T:C \""..cwd..path.."\"", "r" )
```
This smells like an RCE in the making.